react/features/device-selection/components/DeviceSelection.js

```diff
@@ -0,0 +1,586 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import {
+    setAudioInputDevice,
+    setAudioOutputDevice,
+    setVideoInputDevice
+} from '../../base/devices';
+import {
+    Dialog,
+    hideDialog
+} from '../../base/dialog';
+import { translate } from '../../base/i18n';
+import {
+    createLocalAudioTrackPreview,
+    createLocalVideoTrackPreview
+} from '../../base/lib-jitsi-meet';
+
+import AudioInputPreview from './AudioInputPreview';
+import AudioOutputPreview from './AudioOutputPreview';
+import DeviceSelector from './DeviceSelector';
+import VideoInputPreview from './VideoInputPreview';
+
+/**
+ * React component for previewing and selecting new audio and video sources.
+ *
+ * @extends Component
+ */
+class DeviceSelection extends Component {
```

Comment:
Let's name this one DeviceSelectionDialog, so that it's clear that it's the dialog component and is easy to distinguish it from DeviceSelector component